File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    cloth = {}
    for i in clothes:
        if i[1] not in cloth:
            cloth[i[1]] = 1
        else: cloth[i[1]] += 1
        
    answer = 1
    for a, b in cloth.items():
        answer *= (b + 1)
    
    return answer - 1;

File: 프로그래머스/lv2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    unordered_map<string, int> cloth;
    
    for (const auto& item : clothes) {
        cloth[item[1]]++;
    }
    
    int answer = 1;
    for (const auto& pair : cloth) {
        answer *= (pair.second + 1);
    }
    
    return answer - 1;
}